kroxylicious-operator/src/main/java/io/kroxylicious/operator/ProxyConfigSecret.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.kroxylicious.operator;
+
+import java.util.Map;
+
+import io.fabric8.kubernetes.api.model.Secret;
+import io.fabric8.kubernetes.api.model.SecretBuilder;
+import io.javaoperatorsdk.operator.api.reconciler.Context;
+import io.javaoperatorsdk.operator.processing.dependent.kubernetes.CRUDKubernetesDependentResource;
+import io.javaoperatorsdk.operator.processing.dependent.kubernetes.KubernetesDependent;
+
+import io.kroxylicious.crapi.v1alpha1.KafkaProxy;
+
+/**
+ * A Kube {@code Secret} containing the proxy config YAML.
+ * We use a {@code Secret} (rather than a {@code ConfigMap})
+ * because the config might contain sensitive settings like passwords
```

Comment:
I'm fine with this a bootstrapping choice, but I think I'd rather explore using a config map and having some way to embed secrets (well references to them) in it.

Reply:
My other thought which possible re-opens the old wounds of your API thinking is should we even have the config a single entity in the API Server? Could we not just generate a file on the fly for mounting into the container and present higher level entities in the API Server?

Again this works for me as a bootstrapping option.

Reply:
> should we even have the config a single entity in the API Server?

I'm keen to at least _try_ to have a CRD per plugin model, if that's what you mean? But not in this PR.

> I think I'd rather explore using a config map and having some way to embed secrets (well references to them) in it.

The tricky thing about that is it forces the operator into knowing about all those secrets and doing all that mounting. I don't doubt that it could be done. But it's not clear what benefits it brings over just mounting one big secret. Can you seem some concrete advantages?

Of course there's the parallel problem of needing to refer to sensitive info from the CR, and how that's expressed in the CRD validation schema. I guess it we can do that nicely then we're a step closer to being able to do what you want.

Reply:
From my experience, it's easiest for operation to have a config map and secrets separated.
Secrets can come from another context (and responsibility) than the config.
In a concret case, there can have a team (A) managing the kroxylicious deployment and another (B) managing kafka, only required secret is coming from team B, it doesn't make sens to ask this team to manage config coming from team A.